test/utils/fin.ts

```diff
@@ -0,0 +1,172 @@
+import {EventEmitter} from 'events';
+
+import {Identity} from 'hadouken-js-adapter';
+import * as deepmerge from 'deepmerge';
+
+const emitter = new EventEmitter();
+const apps = new Map<String, FakeApplication>();
+
+export const fin = {
+    System: {
+        addListener: (event, listener) => {
+            emitter.addListener(event, listener);
+        },
+        removeListener: (event, listener) => {
+            emitter.removeListener(event, listener);
+        },
+        getAllApplications: async () => {
+            return Array.from(apps.values()).map(app => {
+                return {
+                    isRunning: app.state.isRunning,
+                    uuid: app.identity.uuid,
+                    parentUuid: app.parentUuid
+                };
+            });
+        }
+    },
+    Application: {
+        me: {uuid: '', name: ''},
+        create: (identity: Identity, manifest: any, parentUuid?: string) => {
```

Comment:
I made this mock knowing that it would only ever be used in this module in a very simple way.  I can make it super consistent with `fin` but don't think a great deal of time should be spent on it.

Reply:
Fixed with types and asyncs

Reply:
👍